### ISSUE TITLE
Modulo-based batch cycling duplicates data within epochs

### ISSUE THREAD

**Issue by zeinali72** [zeinali72/swe-pinn#60]
## Description

In `src/data/batching.py`, `get_batches_tensor` (lines 19-31) cycles through batches using modulo:

```python
indices = jnp.arange(total_batches) % n_batches_avail
return data[indices]
```

When `total_batches > n_batches_avail`, later batches are exact duplicates of earlier ones.

## Impact

The network sees repeated data within a single epoch, reducing effective sample diversity and wasting gradient steps.

## Fix

Sample enough points to fill `total_batches * batch_size` directly, or re-permute and tile.

## Category
**Medium** · Pillar: Performance Check
